### PR TITLE
Update README.md for Project Euler

### PR DESCRIPTION
**Pull request by dhruvmanila**
To see what it looks like: https://github.com/dhruvmanila/Python/blob/readme-pe/project_euler/README.md

In line with the following comments:
- https://github.com/TheAlgorithms/Python/pull/3215#issuecomment-706837753
- https://github.com/TheAlgorithms/Python/issues/2786#issuecomment-703365779
- https://github.com/TheAlgorithms/Python/issues/2695#issuecomment-703122388

### **Describe your change:**

* [x] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.

### CODE REVIEW

**Comment by dhruvmanila** [comment]
Hey, @cclauss Can you take a look at this? Thanks if you do.

File: project_euler/README.md [hunk 1]
```diff
@@ -9,3 +9,109 @@ Here the efficiency of your code is also checked.
 I've tried to provide all the best possible solutions.
```
Comment (poyea) [inline project_euler/README.md:9]: Seems weird. Maybe delete L8-9 or rephrase them?

File: project_euler/README.md [hunk 2]
```diff
@@ -9,3 +9,109 @@ Here the efficiency of your code is also checked.
...
 
 For description of the problem statements, kindly visit https://projecteuler.net/show=all
```
Comment (poyea) [inline project_euler/README.md:11]: We don't need this, since we have L3?

File: project_euler/README.md [hunk 3]
```diff
@@ -9,3 +9,109 @@ Here the efficiency of your code is also checked.
...
+
+
+## Solution Guidelines
+
+Welcome to [TheAlgorithms/Python](https://github.com/TheAlgorithms/Python)! Before reading the solution guidelines, make sure you read the whole [contributing guidelines](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md) as it won't be repeated in here. If you have any doubt on the guidelines, please feel free to [state it clearly in an issue](https://github.com/TheAlgorithms/Python/issues/new) or ask the community in [Gitter](https://gitter.im/TheAlgorithms). You can use the [template](https://github.com/TheAlgorithms/Python/blob/master/project_euler/README.md#solution-template) we have provided below as your starting point but be sure to read the [coding style](https://github.com/TheAlgorithms/Python/blob/master/project_euler/README.md#coding-style) part first.
```
Comment (poyea) [inline project_euler/README.md:16]:
```suggestion
Welcome to [TheAlgorithms/Python](https://github.com/TheAlgorithms/Python)! Before reading the solution guidelines, make sure you read the whole [Contributing Guidelines](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md) as it won't be repeated in here. If you have any doubt on the guidelines, please feel free to [state it clearly in an issue](https://github.com/TheAlgorithms/Python/issues/new) or ask the community in [Gitter](https://gitter.im/TheAlgorithms). You can use the [template](https://github.com/TheAlgorithms/Python/blob/master/project_euler/README.md#solution-template) we have provided below as your starting point but be sure to read the [Coding Style](https://github.com/TheAlgorithms/Python/blob/master/project_euler/README.md#coding-style) part first.
```

File: project_euler/README.md [hunk 4]
```diff
@@ -9,3 +9,109 @@ Here the efficiency of your code is also checked.
...
+
+### Coding Style
+
+* Please maintain consistency in project directory and solution file names. Keep the following points in mind:
+  * Create a new directory only for the problems which does not exists yet.
```
Comment (poyea) [inline project_euler/README.md:21]:
```suggestion
  * Create a new directory only for the problems which do not exist yet.
```

File: project_euler/README.md [hunk 5]
```diff
@@ -9,3 +9,109 @@ Here the efficiency of your code is also checked.
...
+  * If you create a new directory, please create an empty `__init__.py` file as well.
```
Comment (poyea) [inline project_euler/README.md:22]:
```suggestion
  * If you create a new directory, please create an empty `__init__.py` file inside it as well.
```

File: project_euler/README.md [hunk 6]
```diff
@@ -9,3 +9,109 @@ Here the efficiency of your code is also checked.
...
+  * Please name the project directory as `problem_<problem_number>` where `problem_number` should be filled with 0s so as to occupy 3 decimal places. Example: `problem_001`, `problem_002`, `problem_067`, `problem_145`, and so on.
```
Comment (poyea) [inline project_euler/README.md:23]:
```suggestion
  * Please name the project directory as `problem_<problem_number>` where `problem_number` should be filled with 0s so as to occupy 3 digits. Example: `problem_001`, `problem_002`, `problem_067`, `problem_145`, and so on.
```

File: project_euler/README.md [hunk 7]
```diff
@@ -9,3 +9,109 @@ Here the efficiency of your code is also checked.
...
+
+* Please provide a link to the problem and other references, if used, in the module-level docstring.
+
+* All imports should come ***after*** the module-level docstring.
+
+* You can have as many helper functions as you want but there should be one main function called `solution` which should satisfy the conditions as stated below:
+  * It should contain positional argument(s) whose default value is the question input. Example: Please take a look at [problem 1](https://projecteuler.net/problem=1) where the question is to *Find the sum of all the multiples of 3 or 5 below 1000.* In this case the main solution function will be `solution(limit: int = 1000)`.
+  * When the `solution` function is called without any arguments like so: `solution()`, it should return the answer to the problem.
+
+* Every function, which includes all the helper functions if any and the main solution function, should have `doctest` in the function docstring along with a brief statement mentioning what the function is about.
```
Comment (poyea) [inline project_euler/README.md:33]:
```suggestion
* Every function, which includes all the helper functions, if any, and the main solution function, should have `doctest` in the function docstring along with a brief statement mentioning what the function is about.
```

File: project_euler/README.md [hunk 8]
```diff
@@ -9,3 +9,109 @@ Here the efficiency of your code is also checked.
...
+      >>> solution(16)
+      ...
+      >>> solution(100)
+      ...
+      """
+    ```
+
+### Solution Template
+
+You can use the below template as your starting point but please read the [coding style](https://github.com/TheAlgorithms/Python/blob/master/project_euler/README.md#coding-style) first to understand how the template works.
```
Comment (poyea) [inline project_euler/README.md:54]:
```suggestion
You can use the below template as your starting point but please read the [Coding Style](https://github.com/TheAlgorithms/Python/blob/master/project_euler/README.md#coding-style) first to understand how the template works.
```

**Review by poyea — APPROVED** [review]
That's cool. I just add a few suggestions on wording, etc. Thank you for your pull request!🤩

**Pull request merged**